### PR TITLE
docs: remove v12 references

### PR DESCRIPTION
**Pull request by stevenGravy**

### CODE REVIEW

File: docs/pages/admin-guides/access-controls/device-trust/device-management.mdx [hunk 1]
```diff
@@ -82,11 +82,8 @@ Asset Tag    OS    Enroll Status Device ID
 ```
 
 <Admonition type="warning" title="Device Role">
-  For clusters created after v13.3.6, Teleport supports preset `device-admin` role to manage devices.
-
-  For clusters created using Teleport v12 or newer, the preset `editor` role has
-  the necessary permissions to manage devices. For clusters created before v12, you may want
-  to create a dedicated [`device-admin`](#dedicated-device-admin-role) role.
+  For clusters created after v13.3.6, Teleport supports preset `device-admin` role to manage devices
```
Comment (zmb3) [inline docs/pages/admin-guides/access-controls/device-trust/device-management.mdx:85]: Might as well remove this too, v13 is EOL.

**Review by ptgott — APPROVED** [review]

File: docs/pages/admin-guides/access-controls/device-trust/device-management.mdx (hunk 1 above)
Comment (stevenGravy) [reply docs/pages/admin-guides/access-controls/device-trust/device-management.mdx]: thanks, please see update

File: docs/pages/admin-guides/access-controls/device-trust/device-management.mdx [hunk 2]
```diff
@@ -230,8 +222,8 @@ Device "c9cbb327-68a8-497e-b820-6a4b2bf58269" removed
 We recommend creating a dedicated `device-admin` role for device inventory
 management.
 
-<Admonition type="tip" title="v13.3.6+">
-  Teleport version 13.3.6 and above has the preset `device-admin` role, which
+<Admonition type="tip" title="Preset Device Administration role">
+  Teleport has the preset `device-admin` role, which
   is a substitute for the role described below.
```
Comment (zmb3) [inline docs/pages/admin-guides/access-controls/device-trust/device-management.mdx:227]: Let's remove the role below. We're asking the reader to create a role that already exists. This is not necessary.

**Review by scottpgallagher — APPROVED** [review]
Approved.

**Pull request merged**